projects/core/src/occ/adapters/checkout/occ-checkout-payment-type.adapter.spec.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { Type } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { ConverterService, PAYMENT_TYPE_NORMALIZER } from '@spartacus/core';
+import { Occ, OccConfig } from '../../index';
+import { OccCheckoutPaymentTypeAdapter } from './occ-checkout-payment-type.adapter';
+
+const MockOccModuleConfig: OccConfig = {
+  backend: {
+    occ: {
+      baseUrl: '',
+      prefix: '',
+    },
+  },
+  context: {
+    baseSite: [''],
+  },
+};
+
+describe('OccCheckoutPaymentTypeAdapter', () => {
+  let service: OccCheckoutPaymentTypeAdapter;
+  let httpMock: HttpTestingController;
+  let converter: ConverterService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        OccCheckoutPaymentTypeAdapter,
+        { provide: OccConfig, useValue: MockOccModuleConfig },
+      ],
+    });
+    service = TestBed.inject(
+      OccCheckoutPaymentTypeAdapter as Type<OccCheckoutPaymentTypeAdapter>
+    );
+    httpMock = TestBed.inject(
+      HttpTestingController as Type<HttpTestingController>
+    );
+    converter = TestBed.inject(ConverterService as Type<ConverterService>);
+
+    spyOn(converter, 'pipeableMany').and.callThrough();
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  describe('loadPaymentypes', () => {
+    it('should return paymentTypes', () => {
+      const paymentTypesList: Occ.PaymentTypeList = {
+        paymentTypes: [
+          {
+            code: 'card',
+            displayName: 'card',
+          },
+          {
+            code: 'account',
+            displayName: 'account',
+          },
+        ],
+      };
+
+      service.loadPaymentTypes().subscribe((result) => {
+        expect(result).toEqual(paymentTypesList.paymentTypes);
```

Comment:
running assertions inside a subscription used to cause issues where the assertion was never called.
I don't know if this is still an issue.
Storing there result of the observable stream in variable and asserting outside the subscription is what I see is done elsewhere.

Reply:
done.